Adding instructions for Java client installation (#670)
graknlabs/docs: 03-client-api/01-java.md

## Installation

### Prerequisites

To use this client, you need a compatible version of TypeDB Server running. Please see the
[Compatibility Table](#version-compatibility) to check what version do you need, depending on the TypeDB server version
being used.

> 

Author: ```suggestion
```

> Maven is expected to be used as a build system for the Java project.

Reviewer: I'm thinking we should just delete this sentence. It sounds like we're "requiring" or at least "expecting" our users to use Maven. The reality is that many use Gradle, and they are most likely accustomed to converting a `pom.xml` specification to a Gradle dependency spec (which is much terser!)

> Maven is expected to be used as a build system for the Java project.

Author: ```suggestion
```

## Installation

### Prerequisites

To use this client, you need a compatible version of TypeDB Server running. Please see the
[Compatibility Table](#version-compatibility) to check what version do you need, depending on the TypeDB server version
being used.

> Hereinafter maven expected to be used as build system for the Java project.

Reviewer: This wording is awkward, can we refine this sentence?

> Hereinafter maven expected to be used as build system for the Java project.

Author: ```suggestion
Maven is expected to be used as a build system for the Java project.
```

> ### Add a repository with TypeDB Java client to maven

Reviewer: **Maven** should always be capitalised.

Author: ```suggestion
### Add a repository with TypeDB Java client to Maven
```

> Add the code below to the `pom.xml` file in your maven project.

Author: ```suggestion
Add the code below to the `pom.xml` file in your Maven project.
```

> <div class="note">
> [Important]
> Make sure to use the version number obtained in the previous section instead of the `{version}` placeholder. 
> 
> Otherwise, you might have the following error: `Could not find artifact com.vaticle.typedb:typedb-client:pom:{version} 
> in repo.vaticle.com`.
> </div>

Reviewer: In general we don't need to state the obvious like this, I think; but given we've already written it, let's just keep it terse.
```suggestion
<div class="note">
[Important]
Be sure to replace the `{version}` placeholder tag with the version of Client Java you want to install.
</div>
```
